lib/commons/dom/visually-contains.js

```diff
@@ -58,4 +73,39 @@ dom.visuallyContains = function(node, parent) {
 	}
 
 	return true;
+}
+
+/**
+ * Checks whether a parent element visually contains its child, either directly or via scrolling.
+ * Assumes that |parent| is an ancestor of |node|.
+ * @method visuallyContains
+ * @memberof axe.commons.dom
+ * @instance
+ * @param {Element} node
+ * @param {Element} parent
+ * @return {boolean} True if node is visually contained within parent
+ */
+dom.visuallyContains = function visuallyContains(node, parent) {
```

Comment:
This should reset the scroll position after the check has finished. See #1951

Reply:
With these changes the code no longer needs to scroll the page to check for color-contrast.